pyccel/ast/builtin_objects/list_functions.py

```diff
@@ -0,0 +1,65 @@
+# coding: utf-8
+#------------------------------------------------------------------------------------------#
+# This file is part of Pyccel which is released under MIT License. See the LICENSE file or #
+# go to https://github.com/pyccel/pyccel/blob/master/LICENSE for full license details.     #
+#------------------------------------------------------------------------------------------#
+
+"""
+The List container has a number of built-in methods that are 
+always available.
+
+This module contains objects which describe these methods within Pyccel's AST.
+"""
+
+from pyccel.ast.internals import PyccelInternalFunction
+from pyccel.ast.datatypes import NativeHomogeneousList
+
+class ListPop(PyccelInternalFunction) :
+    """
+    Represents a call to the .pop() method.
+    
+    Represents a call to the .pop() method which
+    removes the item at the specified index. 
+    The method also returns the removed item.
+
+    Parameters
+    ----------
+    name : TypedAstNode
+        The name of the list.
+
+    index_element : TypedAstNode
+        The current index value for the element to be popped.
+    """
+    __slots__ = ('_dtype','_precision', '_index','_list_variable')
+    _rank = 0
+    _order = None
+    _shape = None
+    _class_type = NativeHomogeneousList()
+    name = 'pop'
+    def __init__(self, name, index_element=None):
+        if index_element:
+            super().__init__(name, index_element)
+        else:
+            super().__init__(name)
+        self._index = index_element
+        self._list_variable = name
```

Comment:
You are saving `name` in an attribute called `_list_variable`. These two names seem to imply different things. Why the difference? In my opinion `list_variable` makes more sense than `name` as you are storing a `TypedAstNode` describing a list not the name of a Variable (see similar discussion here: https://github.com/pyccel/pyccel/pull/1709#discussion_r1485854955)